### PR TITLE
Prepare packages for binding.instance becoming non-nullable

### PR DESCRIPTION
**Pull request by Hixie**
This prepares this repo for flutter/flutter#64830 to land.

The _ambiguate hack is needed to allow this code to run both with the old API and the new API.

No version change: This is a trivial internal change that should have no practical impact and can be rolled into the next update.
No CHANGELOG change: Same.

### CODE REVIEW

**Comment by Hixie** [comment]
@stuartmorgan any idea why the `publishable` check is failing? It looks like I put in the right magic strings in the PR description...

**Comment by stuartmorgan** [comment]
Have you pushed since then? Annoyingly, re-run doesn't get the new PR description (see https://github.com/flutter/flutter/issues/95594)

**Comment by Hixie** [comment]
I think so, but let me try again.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
For posterity:

> No version change: This is a trivial internal change that should have no practical impact and can be rolled into the next update.

This was not correct, and I should have caught it in review. This was a client-affecting change, but because the version change was skipped, for some packages the changes were never published, because no other version-changing changes happened to some of these packages. Which means that when 3.0 came out, clients of some of these packages are getting warnings even though we had landed the fixes for them months ago.

This is precisely why we have a continuous release model, instead of relying on later changes to get client-affecting PRs out at some later point. So we shouldn't have bypassed it.
